cloudaux/aws/iam.py

```diff
@@ -163,6 +163,33 @@ def get_role_instance_profiles(role, client=None, **kwargs):
     ]
 
 
+@sts_conn('iam', service_type='client')
+@rate_limited()
+def get_role_managed_policy_documents(role, client=None, **kwargs):
+    """Retrieve the currently active policy version document for every managed policy that is attached to the role."""
+    policies = get_role_managed_policies(role, **kwargs)
+
+    policy_documents = zip(
+        (policy['name'] for policy in policies),
+        Parallel(n_jobs=20, backend="threading")(
+            delayed(get_managed_policy_document)
+            (policy['arn'], **kwargs) for policy in policies
+            )
+        )
+    policy_documents = dict(policy_documents)
+
+    return policy_documents
+
+
+@sts_conn('iam', service_type='client')
+@rate_limited()
+def get_managed_policy_document(policy_arn, client=None, **kwargs):
+    """Retrieve the currently active (i.e. 'default') policy version document for a policy."""
+    policy_metadata = client.get_policy(PolicyArn=policy_arn)
+    policy_document = client.get_policy_version(PolicyArn=policy_arn,
+                                                VersionId=policy_metadata['Policy']['DefaultVersionId'])
```

Comment:
Is there ever a situation where `Policy` could be `None`, or lacks a `DefaultVersionId`?

Reply:
- `Policy` cannot be None, but it's possible for us to trip over a NotFound exception here in a race condition.
- There will always be a `DefaultVersionId`. A policy is created with a version of v1, which is the initial `DefaultVersionId`. The `DefaultVersionId` can be switched between versions on the policy and there must always be one, and whichever version is designated as 'default' cannot be deleted. This ensures there is always at least one version and there is always a non-null `DefaultVersionId`.